src/components/image.tsx

```diff
@@ -47,77 +79,56 @@ export const ImageCard: React.FC<Props> = ({ pos, searchType, img, showTitle })
       <img
         onDragStart={(e) => {
           const parent = e.currentTarget.parentNode?.parentNode?.parentElement;
-          if (parent?.className.includes("results")) {
-            sessionStorage.setItem("drag-source", "results");
-          } else {
-            sessionStorage.setItem("drag-source", "collage");
-          }
-
-          sessionStorage.setItem("image-title", title);
-          sessionStorage.setItem("image-url", url);
-          sessionStorage.setItem("image-author", author);
+          sessionStorage.setItem(
+            "drag-source",
+            parent?.className.includes(typeOfSource.RESULTS) ? typeOfSource.RESULTS : typeOfSource.COLLEAGE
+          );
+          sessionStorage.setItem(imageSessionKeys.TITLE, title);
+          sessionStorage.setItem(imageSessionKeys.URL, url);
+          sessionStorage.setItem(imageSessionKeys.AUTHOR, author);
         }}
         onDragEnter={(e) => (e.currentTarget.style.opacity = ".5")}
         onDragLeave={(e) => (e.currentTarget.style.opacity = "")}
         onDragOver={(e) => e.preventDefault()}
         onDragEnd={() => {
-          if (sessionStorage.getItem("drag-source") === "results") {
+          if (sessionStorage.getItem("drag-source") === typeOfSource.RESULTS) {
             return;
           }
 
-          // exchange images
-          const t = sessionStorage.getItem("dropped-title") as string;
-          const u = sessionStorage.getItem("dropped-url") as string;
-          const a = sessionStorage.getItem("dropped-author") as string;
-
-          setTitle(t);
-          setUrl(u);
-          setAuthor(a);
+          const t = sessionStorage.getItem(droppedSessionKeys.TITLE);
+          const u = sessionStorage.getItem(droppedSessionKeys.URL);
+          const a = sessionStorage.getItem(droppedSessionKeys.AUTHOR);
 
-          if (typeof pos === "undefined") {
-            return;
+          if (t && u && a) {
+            // exchange images
+            setImage(t, u, a);
```

Comment:
Although this fixes the `null - null` issue, it introduces duplicates when exchanging with a blank image for some reason.

![image](https://user-images.githubusercontent.com/31902745/99277422-90439980-280c-11eb-8bc4-32f50d9272c6.png)

Reply:
Sorry, I couldn't test it because I couldn't connect the API to the FE. I started FE and API with Key, but when I go to localhost: 3000. FE was not connected with api. I didn't see any request for localhost: 5000. 
both servers started correctly .Did I do something wrong?

Reply:
Yeah i got it, i looked to the code and Now i see that i have to search something.